### DISCUSSION TITLE
Krom Not Running, No errors or problems

### DISCUSSION THREAD

**Discussion by Ron5s** [Kode/Kha#1293] [General]
I haven't been able to run Krom via visual studio code for a while.  Im on windows 10, just updated my Kha and Krom bin repository.  Ive also compiled Krom from source code with no luck.  No errors are produced it just does not run.

Ive also tried to do a full build with OpenGL, it compiled but when I ran the program it behaved differently than running from visual studio code via html.  The antialiassing was not working and the mouse interaction was not working all of the time.

Ive deleted both the Kha and Krom_bin repository and re-cloned them with no change.
Thanks for the help.

**Comment by Ron5s** [comment]
I still cant figure out any of the issues listed above because there are no errors or warnings.  The compiled exe has issues with anti aliasing and mouse input.  This last time it produced a kinc.dmp file (attached).  Any help to know what to look for would be appreciated.

Thank you

[kinc.zip](https://github.com/Kode/Kha/files/6011293/kinc.zip)

**Comment by RobDangerous** [comment]
Hardware anti-aliasing is generally not something you should rely on. That not working might be our fault or might be the OpenGL driver's fault. The mouse problem report is not very precise so no idea.
Other than that, will try to figure out what's going wrong when I get to it. Thanks for the dump.

**Comment by Ron5s** [comment]
The mouse issue only happens in the compiled exe.  It does not happen in the html debug. And Krom just doesn't work via vs code anymore, no errors or warning no window pops up.  Ive even compiled from source.  I wish i could give you more information than that.

For AA im using this line of code
System.start({title: "test", width: WIDTH, height: HEIGHT, framebuffer:{samplesPerPixel: 4}}, function (_)

Thank you for the help.

**Comment by RobDangerous** [comment]
Yes, that's the kind of anti-aliasing you should not rely on. But I think I didn't understand you right at first - when you compiled it I thought you were still talking about Krom but you just compiled your Kha program, right? In that case you can just run it in Visual Studio to easily debug crashes. I also don't recommend using OpenGL on Windows. Still can't say anything about "there are some mouse issues".

**Comment by Ron5s** [comment]
Thank you for your help, Ive been tracking down one of my problems with your sugestions.  I fixed the mouse issue by re-writing a chunk of code.  But I still don't know why there was a difference between a compiled version and the HTML debug version.  But Its fixed.

My last issue is that Krom debug still doesn't work via vs code, the window never pops up and does not provide errors.

Additionally I have a couple of questions:
1)  Shaders seem to be an issue with the compiled version to Directx12 and not opengl.  A DirectX compiled version throws this error

Exception thrown: read access violation.
renderTarget->impl.depthStencilDescriptorHeap was nullptr

kinc_g5_command_list_clear(kinc_g5_command_list * list, kinc_g5_render_target * renderTarget, unsigned int flags, unsigned int color, float depth, int stencil) Line 155	C++


2) What would you suggest I use for anti-aliasing?  A screen shader?

Thanks for the help.

**Comment by RobDangerous** [comment]
So what exactly was up with the mouse?
Shaders for Krom on Windows are compiled for D3D11 in vscode because Krom on Windows should be compiled for D3D11 (and it is by default).
Anti-Aliasing is a big topic, much too big to just recommend something without knowing anything about what you're trying to do.

**Reply by Ron5s** [reply]
Im not sure why there was a difference, but a variable ended up being null in the compiled version but it worked perfectly in the HTML debug version.  It could have been a race condition.  I re-wrote a section of code with better logic (I think I need to take time to do this to everything) and it works now.

OK I think I need to install the D3D11 SDK its not showing up in VSCode as a target.  

So for the AA I'm drawing a GUI containing circles and lines etc.  Im just using AA to make these elements look better. The line of code listed above that you said I should not use worked well but stopped working recently.

Thanks again for the help.

**Comment by RobDangerous** [comment]
You do not need the D3D11 SDK. For Kha and Kinc the Windows target - the one which is only called "Windows" - is the Direct3D11 target.
Drawing circles and lines can mean a lot of things. But also a GitHub issue is not the place to start a deeper discussion about anti-aliasing, sorry.

**Reply by Ron5s** [reply]
Ok great thank you. I’m just drawing a circular outline that is the boarder of the filled circle. But I understand your point. Where would be a more appropriate place to discuss AA?  Last time I checked the other forum does not let me log in. Is there a discord server?

**Comment by RobDangerous** [comment]
Oh, this is the right place, I thought this is a GitHub issue, uups. Just open another thread so things are sorted and all is good.
There's also a discord via http://discord.kode.tech

**Reply by Ron5s** [reply]
Excellent, thank you as always for the help.
